mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -114,21 +106,47 @@ class DenseMatrix(val numRows: Int, val numCols: Int, val values: Array[Double])
   require(values.length == numRows * numCols, "The number of values supplied doesn't match the " +
     s"size of the matrix! values.length: ${values.length}, numRows * numCols: ${numRows * numCols}")
 
-  override def toArray: Array[Double] = values
+  override def toArray: Array[Double] = {
```

Comment:
No, the matrix is used in RDDs. It will be error-prone if we change the underlying data. Just want to make sure that in existing code when we call `values`, we check `isTransposed`.